### PR TITLE
Implement Tracking and Storing of Plausible Statistics

### PR DESCRIPTION
**Pull request by Ashesh3**
This PR introduces functionality to track and store statistics from Plausible, an open-source analytics tool.

**Key changes:**

- Addition of Plausible environment variables (`PLAUSIBLE_HOST`, `PLAUSIBLE_SITE_ID`, `PLAUSIBLE_AUTH_TOKEN`) to the base configuration.

- Creation of new models (`Goal`, `GoalProperty`, `GoalEntry`, `GoalPropertyEntry`) to store Plausible statistics in the database.

- Creation of a new task to capture goals from Plausible and store them in the database. This task is scheduled to run daily.

- Addition of the Plausible statistics capturing task to the list of periodic tasks.

- Addition of Plausible environment variables to AWS backend and celery configuration files.

- Migration file for the newly created models.

This feature enhances the application's analytics capabilities by enabling the tracking and storing of key metrics from Plausible

image

image

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/tasks/plausible_stats.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    response = request(
+        "GET",
+        url,
+        headers={
+            "Authorization": "Bearer " + settings.PLAUSIBLE_AUTH_TOKEN,
+        },
+    )
+
+    return response.json()
```
Comment (sainak) [inline care/facility/tasks/plausible_stats.py:43]: - [ ] use request.get()
- [ ] set a timeout for the api call
- [ ] call request.raise_for_status()

File: care/facility/tasks/plausible_stats.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    @property
+    def formatted_name(self):
+        if self == Goals.PAGEVIEW:
+            return "pageview"  # pageview is a reserved goal in plausible
+        return self.name.replace("_", " ").title()
+
+
+def get_goal_stats(plausible_host, site_id, date, goal_name):
+    goal_filter = f"event:name=={goal_name}"
+    url = f"https://{plausible_host}/api/v1/stats/aggregate?site_id={site_id}&filters={goal_filter}&period=day&date={date}&metrics=visitors,events"
```
Comment (sainak) [inline care/facility/tasks/plausible_stats.py:33]: pass query params as dict to the request call

File: care/facility/tasks/plausible_stats.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    response = request(
+        "GET",
+        url,
+        headers={
+            "Authorization": "Bearer " + settings.PLAUSIBLE_AUTH_TOKEN,
+        },
+    )
+
+    return response.json()
```
Comment (sainak) [inline care/facility/tasks/plausible_stats.py:65]: same for this function

File: care/facility/tasks/plausible_stats.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                )
+                for property_statistic in goal_property_stats["results"]:
+                    goal_property_object, _ = GoalProperty.objects.get_or_create(
+                        goal=goal_object,
+                        name=property_name,
+                    )
+                    (
+                        goal_property_event_object,
+                        _,
+                    ) = GoalPropertyEntry.objects.get_or_create(
```
Comment (sainak) [inline care/facility/tasks/plausible_stats.py:119]:
```suggestion
                    goal_property_event_object, _ = GoalPropertyEntry.objects.get_or_create(
```

Comment (Ashesh3) [reply care/facility/tasks/plausible_stats.py:119]: The black formatter would automatically format it to what it is right now. I cannot commit this suggestion 😅

Comment (sainak) [reply care/facility/tasks/plausible_stats.py:119]: Enclose the queryset in perens that should fix it

File: care/facility/tasks/plausible_stats.py (hunk 2 above)
Comment (Ashesh3) [reply care/facility/tasks/plausible_stats.py]: Done

File: care/facility/tasks/plausible_stats.py (hunk 3 above)
Comment (Ashesh3) [reply care/facility/tasks/plausible_stats.py]: Done

File: care/facility/tasks/plausible_stats.py (hunk 4 above)
Comment (Ashesh3) [reply care/facility/tasks/plausible_stats.py:142]: Still unable to format it as per the suggestion, messaged you personally to resolve this.

**Review by rithviknishad — APPROVED** [review]

**Comment by Ashesh3** [comment]
@vigneshhari Got 2 approvals, could you review?

**Pull request merged**